app/src/main/java/appricottsoftware/clarity/fragments/SettingFragment.java

```diff
@@ -24,6 +49,213 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
+        ClarityApp.getRestClient().getEmail(uid, getContext(), new JsonHttpResponseHandler() {
+            @Override
+            public void onSuccess(int statusCode, Header[] headers, JSONObject response) {
+                try {
+                    email.setText("");
+                    email.setHint(response.getString("email"));
+                    email.clearFocus();
+                }
+                catch (Exception e) {
+                    Log.e(TAG, "getOldEmail: ", e);
+                }
+            }
+
+            @Override
+            public void onFailure(int statusCode, Header[] headers, Throwable throwable, JSONObject errorResponse) {
+                Log.e(TAG, "getOldEmail Failed");
+                super.onFailure(statusCode, headers, throwable, errorResponse);
+            }
+        });
+    }
+
+    private void setEmailListener() {
+        email.setOnEditorActionListener(new TextView.OnEditorActionListener() {
+            @Override
+            public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
+                if (RegisterActivity.isValidEmail(email.getText().toString()) && ((actionId == EditorInfo.IME_ACTION_DONE) || ((event.getKeyCode() == KeyEvent.KEYCODE_ENTER) && (event.getAction() == KeyEvent.ACTION_DOWN)))) {
+                    int uid = ClarityApp.getSession(getContext()).getUserID();
+                    ClarityApp.getRestClient().updateEmail(uid, email.getText().toString(), getContext(), new JsonHttpResponseHandler() {
+                        @Override
+                        public void onSuccess(int statusCode, Header[] headers, JSONObject response) {
+                            getOldEmail();
+                            Toast.makeText(getContext(), "Email updated!", Toast.LENGTH_LONG).show();
+                            InputMethodManager imm = (InputMethodManager)getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
+                            imm.hideSoftInputFromWindow(email.getWindowToken(), 0);
+                            email.clearFocus();
+                        }
+
+                        @Override
+                        public void onFailure(int statusCode, Header[] headers, Throwable throwable, JSONObject errorResponse) {
+                            Log.e(TAG, "Unable to update email ", throwable);
+                            Toast.makeText(getContext(), "Email not updated", Toast.LENGTH_LONG).show();
+                            getOldEmail();
+                            super.onFailure(statusCode, headers, throwable, errorResponse);
+                        }
+                    });
+                    return true;
+                }
+                else {
+                    Toast.makeText(getContext(), "Invalid email", Toast.LENGTH_LONG).show();
+                    return false;
+                }
+            }
+        });
+    }
+
+    private boolean validatePassword(String password) {
```

Comment:
If we care enough, we should mandate that users meet a criteria for safe password checking. I don't care for that to be done on t hisbranch, but it should be done eventually.